tsl/src/continuous_aggs/create.c

```diff
@@ -1529,27 +1560,41 @@ mattablecolumninfo_addentry(MatTableColumnInfo *out, Node *input, int original_q
 			}
 			else
 			{
-				out->mat_groupcolname_list = lappend(out->mat_groupcolname_list, pstrdup(colname));
+				if (!*skip_adding && ((TargetEntry *) input)->ressortgroupref > 0)
+					out->mat_groupcolname_list =
+						lappend(out->mat_groupcolname_list, pstrdup(colname));
```

Comment:
Does this do anything for the finalized form, other than adding the time_bucket column?

Reply:
The `mat_groupcolname_list` list is used for adding indexes to the materialization hypertable.